codalab/rest/account.py

```diff
@@ -80,18 +80,21 @@ def do_signup():
 
     errors = []
 
-    url = 'https://www.google.com/recaptcha/api/siteverify?secret={}&response={}'.format(
-        os.environ['CODALAB_RECAPTCHA_SECRET_KEY'], token
-    )
-    res = requests.post(url)
+    if not token:
+        errors.append('Google reCAPTCHA token is missing.')
+    else:
+        url = 'https://www.google.com/recaptcha/api/siteverify?secret={}&response={}'.format(
+            os.environ['CODALAB_RECAPTCHA_SECRET_KEY'], token
+        )
+        res = requests.post(url)
```

Comment:
Instead of using format strings to add in the url parameters, let's pass them in as a dictionary to the params keyword in requests.post (so that they'll be automatically URL-encoded)

Reply:
Updated.